engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayProcessor.java

```diff
@@ -93,8 +93,8 @@ private Either<Failure, List<ExecutableSequenceFlow>> findSequenceFlowsToTake(
       return Either.right(null);
     }
 
-    if (element.getOutgoing().size() == 1) {
-      // only one flow can just be taken
+    if (element.getOutgoing().size() == 1 && element.getOutgoing().get(0).getCondition() == null) {
+      // only one flow without a condition, can just be taken
```

Comment:
🔧 Small update
```suggestion
      // only one flow without a condition can be taken
```

Reply:
Let's keep the status quo. 😄 

https://github.com/camunda/zeebe/blob/26a44bdeaec381c988baee50a7e80ca0de7a48fb/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/gateway/ExclusiveGatewayProcessor.java#L93-L94